### PR TITLE
impl(pubsub): add scheduler span

### PR DESCRIPTION
**Pull request by alevenberg**
Closes [#13722](https://github.com/googleapis/google-cloud-cpp/issues/13722)

This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13914)

### CODE REVIEW

File: google/cloud/pubsub/internal/tracing_batch_callback.cc [hunk 1]
```diff
@@ -155,6 +156,34 @@ class TracingBatchCallback : public BatchCallback {
...
+    auto spans = spans_by_ack_id_.find(ack_id);
+    if (spans != spans_by_ack_id_.end()) {
+      auto subscribe_span = spans->second.subscribe_span;
+      if (subscribe_span) {
+        opentelemetry::trace::StartSpanOptions options;
+        options.parent = subscribe_span->GetContext();
+        auto span =
+            internal::MakeSpan("subscriber scheduler",
+                               {{sc::kMessagingSystem, "gcp_pubsub"}}, options);
+        spans->second.scheduler_span = std::move(span);
```
Comment (devbww) [inline google/cloud/pubsub/internal/tracing_batch_callback.cc:171]: Nit: `auto x = exp; y = std::move(x);` could just be `y = exp`.

File: google/cloud/pubsub/internal/tracing_batch_callback.cc [hunk 2]
```diff
@@ -93,6 +93,7 @@ class TracingBatchCallback : public BatchCallback {
       if (kv.second.subscribe_span) kv.second.subscribe_span->End();
       if (kv.second.concurrency_control_span)
         kv.second.concurrency_control_span->End();
```
Comment (devbww) [inline google/cloud/pubsub/internal/tracing_batch_callback.cc:95]: Nit from the past:  We typically use `{}`s when the entire `if` doesn't fit on one line.

**Review by devbww — APPROVED** [review]

File: google/cloud/pubsub/internal/tracing_batch_callback.cc (hunk 2 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_batch_callback.cc:95]: Done.

File: google/cloud/pubsub/internal/tracing_batch_callback.cc [hunk 3]
```diff
@@ -135,10 +137,9 @@ class TracingBatchCallback : public BatchCallback {
       if (subscribe_span) {
         opentelemetry::trace::StartSpanOptions options;
         options.parent = subscribe_span->GetContext();
-        auto span =
-            internal::MakeSpan("subscriber concurrency control",
-                               {{sc::kMessagingSystem, "gcp_pubsub"}}, options);
-        spans->second.concurrency_control_span = std::move(span);
+        spans->second.concurrency_control_span = std::move(internal::MakeSpan(
```
Comment (devbww) [inline google/cloud/pubsub/internal/tracing_batch_callback.cc:140]: You don't need the `move` now because the argument is already a temporary.

Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_batch_callback.cc]: Oops. Fixed.

**Pull request merged**